Update documentation for doc_cfg feature (#92818)
rust-lang/rust: src/doc/rustdoc/src/unstable-features.md

`#![feature(doc_cfg)]` feature gate. For more information, see [its chapter in the Unstable
Book][unstable-doc-cfg] and [its tracking issue][issue-doc-cfg].

> Something important to be noted is that since [#89596](https://github.com/rust-lang/rust/pull/89596)

Reviewer: Hmm, I'm not sure if we should be referencing PRs in the docs. It doesn't really add much.

was merged, you don't need to add `#[doc(cfg(..)]` anymore unless you want to override the default
behaviour. So if we take the previous source code:

```rust

> #![feature(doc_cfg)]

Reviewer 2: I believe this behavior will only be enabled if `#![feature(doc_auto_cfg)]` is used.

Author: You're absolutely right. Thanks!

/// Token struct that can only be used on Windows.
#[cfg(any(windows, doc))]
pub struct WindowsToken;

/// Token struct that can only be used on Unix.
#[cfg(any(unix, doc))]
pub struct UnixToken;

/// Token struct that is only available with the `serde` feature
#[cfg(feature = "serde")]
#[derive(serde::Deserialize)]
pub struct SerdeToken;
```

> It'll render almost the same. The difference being that `doc` will also be displayed. To fix this,

Reviewer: ```suggestion
It'll render almost the same, the difference being that `doc` will also be displayed. To fix this,
```

`#![feature(doc_cfg)]` feature gate. For more information, see [its chapter in the Unstable
Book][unstable-doc-cfg] and [its tracking issue][issue-doc-cfg].

> 

Reviewer: I think we should maybe put `doc_auto_cfg` in its own section (perhaps called `#![feature(doc_auto_cfg)]` or something like that). It's kind of a freestanding feature—i.e., you can use it without using `doc_cfg`.

Make sure to change "An extension to this feature" to "An extension to `doc_cfg`".

An extension to this feature called `doc_auto_cfg` was added. With it, you don't need to add
`#[doc(cfg(..)]` anymore unless you want to override the default behaviour. So if we take the
previous source code:

```rust
#![feature(doc_auto_cfg)]

/// Token struct that can only be used on Windows.
#[cfg(any(windows, doc))]
pub struct WindowsToken;

/// Token struct that can only be used on Unix.
#[cfg(any(unix, doc))]
pub struct UnixToken;

/// Token struct that is only available with the `serde` feature
#[cfg(feature = "serde")]
#[derive(serde::Deserialize)]
pub struct SerdeToken;
```

It'll render almost the same, the difference being that `doc` will also be displayed. To fix this,

> you can use `doc_cfg_hide`:

Reviewer: Not a comment about the docs, but shouldn't hiding `doc` be the default behavior?

Author: Once https://github.com/rust-lang/rust/pull/92856 is merged, it'll be pretty easy to ignore it as well.

`#![feature(doc_cfg)]` feature gate. For more information, see [its chapter in the Unstable
Book][unstable-doc-cfg] and [its tracking issue][issue-doc-cfg].

### `doc_auto_cfg`: Automatically generate `#[doc(cfg)]`

> An extension to the `#[doc(cfg)]` feature called `doc_auto_cfg` was added. With it, you don't need
> to add `#[doc(cfg(..)]` anymore unless you want to override the default behaviour. So if we take the

Reviewer: Changes:

- don't say it "was added"; just say what it is (more consistent with other docs)
- replace `..` with `...` since `..` looks like range syntax

```suggestion
`doc_auto_cfg` is an extension to the `#[doc(cfg)]` feature. With it, you don't need
to add `#[doc(cfg(...)]` anymore unless you want to override the default behaviour. So if we take the
```

Author: Done!
